lib/livebook_web/live/settings_live/env_var_component.ex

```diff
@@ -0,0 +1,92 @@
+defmodule LivebookWeb.SettingsLive.EnvVarComponent do
+  use LivebookWeb, :live_component
+
+  alias Livebook.Settings
+  alias Livebook.Settings.EnvVar
+
+  @impl true
+  def update(assigns, socket) do
+    {env_var, operation} =
+      if assigns.env_var,
+        do: {assigns.env_var, :edit},
+        else: {%EnvVar{}, :new}
+
+    changeset = Settings.change_env_var(env_var)
+
+    {:ok,
+     socket
+     |> assign(assigns)
+     |> assign(changeset: changeset, env_var: env_var, operation: operation)}
+  end
+
+  @impl true
+  def render(assigns) do
+    ~H"""
+    <div class="p-6 flex flex-col space-y-5">
+      <h3 class="text-2xl font-semibold text-gray-800">
+        <%= if @operation == :new, do: "Add environment variable", else: "Edit environment variable" %>
+      </h3>
+      <p class="text-gray-700">
+        Configure your application global environment variables.
+      </p>
+      <.form
+        id={"#{@id}-form"}
+        let={f}
+        for={@changeset}
+        phx-target={@myself}
+        phx-submit="save"
+        phx-change="validate"
+        autocomplete="off"
+        spellcheck="false"
+      >
+        <div class="flex flex-col space-y-4">
+          <.input_wrapper form={f} field={:key} class="flex flex-col space-y-1">
+            <div class="input-label">
+              Key <span class="text-xs text-gray-500">(alphanumeric and underscore)</span>
+            </div>
+            <%= text_input(f, :key, class: "input") %>
+          </.input_wrapper>
+          <.input_wrapper form={f} field={:value} class="flex flex-col space-y-1">
+            <div class="input-label">Value</div>
+            <%= text_input(f, :value, class: "input") %>
+          </.input_wrapper>
+
+          <div class="flex space-x-2">
+            <%= submit("Save",
+              class: "button-base button-blue",
+              disabled: not @changeset.valid?,
+              phx_disabled_with: "Adding..."
+            ) %>
+            <button phx-click="cancel" phx-target={@myself} class="button-base button-outlined-gray">
+              Cancel
+            </button>
```

Comment:
it looks like until the Save buttons is enabled, Cancel button is the one activated when pressing Enter. Not sure if this is a big deal in practice but it is unexpected. For example, clicking "Enter" in this state:

<img width="511" alt="image" src="https://user-images.githubusercontent.com/76071/189212097-ccb096d4-d4e0-46f3-b68d-cf6bcf8659e0.png">


would close the modal.

Reply:
So, since we're checking the `socket.assigns.changeset.valid?` inside the save event, we don't need to keep the `Save` button disabled

Reply:
The button being disabled is a good indicator whether the form is good, so we should keep that.

I don't think the button has focus, because on the screenshot it's the input being focused. Perhaps the button is treated as submit and enter triggers it? In that case adding `type="button"` should help.

Reply:
@jonatanklosko good catch, changing here 🏃🏻